packages/devtools_app/assets/scripts/inspector_polyfill_script.dart

```diff
@@ -0,0 +1,244 @@
+//
+// We current execute this library using eval but that should be viewed as an
+// implementation detail as eval is simply the most expedient way to currently
+// late bind these service extensions. It could make sense to provide an
+// alternate eval like mechanism to inject additional libraries into an already
+// running Dart application. For example, the fact that we cannot define
+// additional classes using eval is a little limiting to what we can do within
+// this polyfill.
+//
+// Only code between the INSPECTOR_POLYFILL_SCRIPT_START and
+// the INSPECTOR_POLYFILL_SCRIPT_END tags are actually executed. All other code
+// exists purely to get consistent static warnings with what would be displayed
+// when this code is evaluated in the context of
+// package:flutter/src/widgets/widget_inspector.dart
+//
+// The imports in this file intentionally match the imports specified by
+// package:flutter/src/widgets/widget_inspector.dart so that we get consistent
+// static warnings and autocompletes while editing this file. Do not add
+// additional imports to this file.
+// ignore_for_file: unused_import
+
+// Code from this class is executed within the context of the
+// widget_inspector.dart library so using protected members is fine.
+// ignore_for_file: invalid_use_of_visible_for_testing_member
+// ignore_for_file: invalid_use_of_protected_member
+
+import 'dart:async';
+import 'dart:convert';
+import 'dart:developer' as developer;
+import 'dart:math' as math;
+import 'dart:typed_data';
+
+import 'dart:ui' as ui
+    show
+        ClipOp,
+        Image,
+        ImageByteFormat,
+        Paragraph,
+        Picture,
+        PictureRecorder,
+        PointMode,
+        SceneBuilder,
+        Vertices;
+import 'dart:ui' show Canvas, Offset;
+
+import 'package:flutter/foundation.dart';
+import 'package:flutter/painting.dart';
+import 'package:flutter/rendering.dart';
+import 'package:flutter/scheduler.dart';
+import 'package:vector_math/vector_math_64.dart';
+import 'package:flutter/src/widgets/app.dart';
+import 'package:flutter/src/widgets/basic.dart';
+import 'package:flutter/src/widgets/binding.dart';
+import 'package:flutter/src/widgets/debug.dart';
+import 'package:flutter/src/widgets/framework.dart';
+import 'package:flutter/src/widgets/gesture_detector.dart';
+import 'package:flutter/src/widgets/widget_inspector.dart';
+
+// This extension method is simply a hack to make a private member defined on
+// WidgetInspectorService show up as in scope as far as the analysis server is
+// concerned.
+extension WidgetInspectorServicePrivateMethods on WidgetInspectorService {
+  Map<String, Object> _nodeToJson(
+    DiagnosticsNode node,
+    InspectorSerializationDelegate delegate,
+  ) {
+    throw 'Dummy extension method to make the code type check when it calls private members';
+  }
+}
+
+void addServiceExtensions() {
+  // INSPECTOR_POLYFILL_SCRIPT_START
+  T toEnumEntry<T>(List<T> enumEntries, String name) {
```

Comment:
None of this code is new. It is just moved from the existing code snippets within source code that @adalberht already defined.